src/Neo/SmartContract/Manifest/ExtendedType.cs

```diff
@@ -0,0 +1,331 @@
+        /// (these MUST use namedtype referring to a valid type specified in the namedtypes object).
+        /// </summary>
+        public ContractParameterDefinition[]? Fields { get; set; }
+
+        void IInteroperable.FromStackItem(StackItem stackItem)
+        {
+            FromStackItem((VM.Types.Array)stackItem, 0);
+        }
+
+        internal void FromStackItem(VM.Types.Array array, int startIndex)
+        {
+            Type = (ContractParameterType)(byte)array[startIndex++].GetInteger();
+            if (!Enum.IsDefined(typeof(ContractParameterType), Type)) throw new FormatException();
+            NamedType = array[startIndex++].GetString();
+
+            if (array[startIndex++] is Integer length)
+            {
+                Length = checked((int)length.GetInteger());
+                if (Length < 0) throw new FormatException("Length must be non-negative.");
+                if (Length > ExecutionEngineLimits.Default.MaxItemSize) throw new FormatException($"Length must less than {ExecutionEngineLimits.Default.MaxItemSize}.");
+            }
+            else
+            {
+                Length = null;
+            }
+
+            if (array[startIndex++] is VM.Types.Boolean forbidnull)
+            {
+                ForbidNull = forbidnull.GetBoolean();
+            }
+            else
+            {
+                ForbidNull = null;
+            }
+
+            if (array[startIndex++] is ByteString interf)
+            {
+                if (!Enum.TryParse<Nep25Interface>(interf.GetString(), false, out var inferValue))
+                    throw new FormatException();
+
+                Interface = inferValue;
+            }
+            else
+            {
+                Interface = null;
+            }
+
+            if (array[startIndex++] is ByteString key)
+            {
+                if (!Enum.TryParse<ContractParameterType>(key.GetString(), false, out var keyValue))
+                    throw new FormatException();
+
+                Key = keyValue;
+            }
+            else
+            {
+                Key = null;
+            }
+
+            if (array[startIndex++] is Struct value)
+            {
+                Value = new ExtendedType();
+                Value.FromStackItem(value, 0);
+            }
+            else
+            {
+                Value = null;
+            }
+
+            if (array[startIndex++] is VM.Types.Array fields)
+            {
+                Fields = new ContractParameterDefinition[fields.Count];
+                for (var i = 0; i < fields.Count; i++)
+                {
+                    var field = new ContractParameterDefinition();
+                    field.FromStackItem((VM.Types.Array)fields[i]);
+                    Fields[i] = field;
+                }
+            }
+            else
+            {
+                Fields = null;
+            }
+        }
```

Comment:
I think we need to add a verification method that will be used by both `FromStackItem` and `FromJson`. This verification method should check that the combination of non-Null fields is valid.

For example, according to NEP-25 `NamedType` can't be combined with other fields except `Type`. There are other similar cases, we need to check all of them during deserialization.

Reply:
@AnnaShaleva this property it's also optional in the proposal